src/main/java/ch/njol/skript/expressions/base/SimplePropertyExpression.java

```diff
@@ -28,33 +28,47 @@
 
 /**
  * A base class for property expressions that requires only few overridden methods
- * 
- * @author Peter Güttinger
+
  * @see PropertyExpression
  * @see PropertyExpression#register(Class, Class, String, String)
  */
 public abstract class SimplePropertyExpression<F, T> extends PropertyExpression<F, T> implements Converter<F, T> {
-	
-	@SuppressWarnings({"unchecked", "null"})
+
+	private Event event;
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr((Expression<? extends F>) exprs[0]);
 		return true;
 	}
-	
+
 	protected abstract String getPropertyName();
-	
+
 	@Override
 	@Nullable
 	public abstract T convert(F f);
-	
+
 	@Override
-	protected T[] get(final Event e, final F[] source) {
+	protected T[] get(Event event, F[] source) {
+		this.event = event;
 		return super.get(source, this);
 	}
-	
+
+	/**
+	 * Returns the Event that was used in this expression.
+	 * Will only be null if called in the init method.
+	 * 
+	 * @return The Event that was used in this expression.
+	 */
+	@Nullable
+	public Event getEvent() {
```

Comment:
If there's some debate about whether this should be available you could at least make it protected so that only its inheritors can see it.